firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionTesterApiClient.java

```diff
@@ -55,38 +68,57 @@ class FirebaseAppDistributionTesterApiClient {
   private static final String DOWNLOAD_URL_KEY = "downloadUrl";
 
   private static final String TAG = "TesterApiClient:";
+  private static final int DEFAULT_BUFFER_SIZE = 8192;
 
-  public static final int DEFAULT_BUFFER_SIZE = 8192;
-
+  private final FirebaseApp firebaseApp;
+  private final Provider<FirebaseInstallationsApi> firebaseInstallationsApiProvider;
   private final HttpsUrlConnectionFactory httpsUrlConnectionFactory;
+  private final Executor taskExecutor;
 
-  FirebaseAppDistributionTesterApiClient() {
-    this(new HttpsUrlConnectionFactory());
+  FirebaseAppDistributionTesterApiClient(
+      @NonNull FirebaseApp firebaseApp,
+      @NonNull Provider<FirebaseInstallationsApi> firebaseInstallationsApiProvider) {
+    this(
+        Executors.newSingleThreadExecutor(),
+        firebaseApp,
+        firebaseInstallationsApiProvider,
+        new HttpsUrlConnectionFactory());
   }
 
   @VisibleForTesting
   FirebaseAppDistributionTesterApiClient(
+      @NonNull Executor taskExecutor,
```

Comment:
Is this ever anything other than `Executors.newSingleThreadExecutor()`? It seems that both, production and test code, pass in the same thing, but maybe I'm overlooking a caller.

If it's always the same, remove it as a parameter.

Reply:
This constructor is here, and `@VisibleForTesting`, because we want to pass in an executor from the test, so that we can await the async work that's enqueued to it.